geopandas/geodataframe.py

```diff
@@ -723,9 +724,44 @@ def to_json(self, na="null", show_bbox=False, drop_id=False, **kwargs):
         GeoDataFrame.to_file : write GeoDataFrame to file
 
         """
-        return json.dumps(
-            self._to_geo(na=na, show_bbox=show_bbox, drop_id=drop_id), **kwargs
+
+        if date_format not in {"epoch", "iso", None}:
+            raise ValueError("date_format must be one of 'epoch', 'iso' or None/omitted (default 'epoch')")
+        if date_format not in {"epoch", "iso", None}:
+            raise ValueError("extra_geom_format must be or one of 'wkb', 'wkt' or None/omitted (default 'wkb')")
+
+        if date_format is None:
+            date_format = "epoch"
+        if extra_geom_format is None:
+            extra_geom_format = "wkb"
+
+        # Detect columns with problematic datatypes and handle them accordingly
+        problem_cols = []
+        for col in self.columns:
+            if is_datetime64_any_dtype(self[col].dtype):
+                # Can't get a virtual column to work!
+                # self.assign(newcol=(self[col] - pd.Timestamp("1970-01-01")) // pd.Timedelta('1s'))
+                problem_cols.append(col)
+                self.rename(columns={col: f"__ORIGINAL__{col}"}, inplace=True)
+                if date_format == "epoch":
+                    self[col] = self[f"__ORIGINAL__{col}"].apply(lambda x: (x - pd.Timestamp("1970-01-01")) // pd.Timedelta('1s'))
```

Comment:
```suggestion
                    self[col] = (self[f"__ORIGINAL__{col}"] - pd.Timestamp("1970-01-01")) // pd.Timedelta('1s')
```
No need for apply here I think.